### PR TITLE
Update to React ^16

### PR DESCRIPTION
**Pull request by davegomez**
Update Britecharts React to use and support React ^16

## Description

With React 16 the library is moving to a new era of better DOM manipulation - a lot of new ways to handle the state, the lifecycle methods, and the errors; and great additional gains in performance.

Britecharts React should be able to get advantage of all of these new features.

**Note:** this PR address issue #78

## Motivation and Context

React ^16 provides a lot of new features and bug fixes that make it the best version of the library so far. With these changes we will be able to write more succinct and better performing code, as well reduce the bundle size in the future.

## How Has This Been Tested?

The whole project has been tested, including all of the scripts in the `package.json` file, using two different prototype projects using React 15.x and React 16.x.

### How to Test
Clone the Britecharts React repository from my fork, install the dependencies, and link the project to your global Node installation:
```
git clone git@github.com:davegomez/britecharts-react.git
cd britecharts-react
git checkout feat-react-16
yarn install
yarn link
```

Now you can clone any of the testing projects, install the dependencies, link your local Britecharts React version, and run the project:
```
git clone git@github.com:davegomez/britecharts-react-15.git
  /* or */
git clone git@github.com:davegomez/britecharts-react-16.git

cd britecharts-react-1x
yarn install
yarn link britecharts-react
yarn start
```

## Types of changes

- [x] Refactor (changes the way we code something without changing its functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by davegomez** [comment]
It seems the build failed because of some code features not supported by Node 6.x but at the same time are being marked as errors by Eventbrite ESLint configuration. Any comments or positions about this?

File: src/charts/loading/LoadingContainer.js [hunk 1]
```diff
@@ -28,6 +28,9 @@ export default class LoadingContainer extends PureComponent {
             ]
         ),
         loadingState: PropTypes.string,
+        height: PropTypes.number,
+        width: PropTypes.number,
+        shouldShowLoadingState: PropTypes.bool,
```
Comment (Golodhros) [inline src/charts/loading/LoadingContainer.js]: Thanks a lot!

File: package.json [hunk 2]
```diff
@@ -63,53 +63,57 @@
...
+    "react-styleguidist": "^8.0.3",
+    "react-test-renderer": "^16.6.1",
     "rimraf": "^2.6.2",
-    "webpack": "^3.6.0",
-    "webpack-bundle-analyzer": "^2.9.0",
-    "webpack-dashboard": "1.0.2",
-    "webpack-dev-server": "^2.8.2",
-    "webpack-merge": "^4.1.0"
+    "uglifyjs-webpack-plugin": "^2.0.1",
+    "webpack": "^4.25.1",
```
Comment (Golodhros) [inline package.json]: Wow, thanks a lot!

**Comment by Golodhros** [comment]
Hi @davegomez, 

thanks a lot for this!! A lot of work and greatly implemented!

I cannot see the errors in the build logs. We should definitely update our node version ASAP. I got back from our OPS team and they will get to the release issue pretty soon.

In the meanwhile, please do whatever necessary (maybe adding a comment) in order to get this merged. Later I could update node and undo it.

Thanks again!!

File: README.md [hunk 3]
```diff
@@ -8,10 +8,10 @@
...
 | ------------- | ------------- | ------------- |
-| [![Donut Chart][donutChartImg]][donutChartDemo] | [![Legend][legendImg]][legendDemo] | [![Tooltip][tooltipImg]][tooltipDemo] |
-| [![Sparkline Chart][sparklineImg]][sparklineDemo] |
+| [![Donut Chart](https://raw.githubusercontent.com/eventbrite/britecharts-react/master/src/docs/images/thumbnails/donut-chart.png)](https://eventbrite.github.io/britecharts-react/#donut) | [![Legend](https://raw.githubusercontent.com/eventbrite/britecharts-react/master/src/docs/images/thumbnails/legend.png)](https://eventbrite.github.io/britecharts-react/#legend) | [![Tooltip](https://raw.githubusercontent.com/eventbrite/britecharts-react/master/src/docs/images/thumbnails/legend.png)](https://eventbrite.github.io/britecharts-react/#tooltip) |
+| [![Sparkline Chart](https://raw.githubusercontent.com/eventbrite/britecharts-react/master/src/docs/images/thumbnails/sparkline.png)](https://eventbrite.github.io/britecharts-react/#sparkline) |
```
[the remark this answers is not in the document]
Comment (davegomez) [reply README.md]: I wish but it seems there is an issue or change with the new `react-styleguidist` version an is unable to render the short names =(

**Comment by davegomez** [comment]
Sure @Golodhros let me know if I need to modify anything else in order for the build to pass.

I really tried to used the short names in the MD file but I don't understand why the new `react-styleguidist` is not parsing that syntax.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Golodhros) [reply README.md]: Oh, I see. 
Styleguidist hasn't been working that well. I tried to update it but it broke a lot of things. We are thinking about moving out of it.

**Comment by davegomez** [comment]
Done @Golodhros, this PR is updated with your change over the CI file. Thanks!

**Comment by Golodhros** [comment]
Yay!!
Thanks a lot @davegomez!! This was a great job!!

**Pull request merged**